Content.Server/Fluids/EntitySystems/PuddleSystem.cs

```diff
@@ -0,0 +1,295 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using Content.Server.Fluids.Components;
+using Content.Shared.Chemistry.EntitySystems;
+using Content.Shared.Chemistry.Reagent;
+using Content.Shared.Directions;
+using Content.Shared.Examine;
+using Content.Shared.Fluids;
+using Content.Shared.Maps;
+using Content.Shared.Physics;
+using Content.Shared.Slippery;
+using JetBrains.Annotations;
+using Robust.Server.GameObjects;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Localization;
+using Robust.Shared.Map;
+using Robust.Shared.Maths;
+using Robust.Shared.Physics;
+using Robust.Shared.Random;
+
+namespace Content.Server.Fluids.EntitySystems
+{
+    [UsedImplicitly]
+    public partial class PuddleSystem : EntitySystem
+    {
+        [Dependency] private readonly IMapManager _mapManager = default!;
+        [Dependency] private readonly IRobustRandom _random = default!;
+        [Dependency] private readonly SolutionContainerSystem _solutionContainerSystem = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            _mapManager.TileChanged += HandleTileChanged;
+
+            SubscribeLocalEvent<PuddleComponent, ExaminedEvent>(HandlePuddleExamined);
+            SubscribeLocalEvent<PuddleComponent, ComponentInit>(OnPuddleInit);
+            SubscribeLocalEvent<PuddleComponent, MapInitEvent>(OnMapInit);
+            SubscribeLocalEvent<PuddleComponent, SolutionChangedEvent>(OnPuddleUpdate);
+        }
+
+        private void OnMapInit(EntityUid uid, PuddleComponent component, MapInitEvent args)
+        {
+            var spriteComponent = ComponentManager.GetComponent<SpriteComponent>(uid);
```

Comment:
As elsewhere please don't GetComponent on this. If the code was already like that then the code was stupid.